### PR TITLE
Tsk1210: Replaced LoginModule with Embedded Spring LDAP

### PR DESCRIPTION
**Pull request by holgerhagen**

### CODE REVIEW

File: rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/RestConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import pro.taskana.common.rest.ldap.LdapClient;
+import pro.taskana.monitor.api.MonitorService;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.workbasket.api.WorkbasketService;
+
+/** Configuration for REST service. */
+@Configuration
+@ComponentScan
+@EnableTransactionManagement
+public class RestConfiguration {
```
Comment (gitgoodjhe) [inline rest/taskana-rest-spring-example-common/src/test/java/pro/taskana/RestConfiguration.java:27]: Do we really need this class here? Can't we use the RestConfiguration from taskana-rest-spring module?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TaskanaSpringBootTest.java [hunk 2]
```diff
@@ -15,6 +17,7 @@
 /** Use this annotation to test with a spring context and a standardized configuration. */
 @Target(ElementType.TYPE)
 @Retention(RetentionPolicy.RUNTIME)
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/TaskanaSpringBootTest.java:20]: We talked about this briefly. What was the reason you had to mark every test as dirty? Can you document this and also state that this is just temporary?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java [hunk 3]
```diff
@@ -1,45 +1,51 @@
...
+  private String ldapBaseDn;
+
+  @Value("${taskana.ldap.groupSearchBase:cn=groups}")
+  private String ldapGroupSearchBase;
+
+  @Value("${taskana.ldap.userDnPatterns:uid={0},cn=users}")
+  private String ldapUserDnPatterns;
+
+  @Value("${taskana.ldap.groupSearchFilter:uniqueMember={0}}")
+  private String ldapGroupSearchFilter;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java:48]: Do we really need the default values?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java [hunk 4]
```diff
@@ -70,20 +76,71 @@ protected void configure(HttpSecurity http) throws Exception {
...
+        .and()
+        .passwordCompare()
+        .passwordAttribute("userPassword");
+  }
+
+  @Bean
+  public LdapAuthoritiesPopulator authoritiesPopulator() {
+    Function<Map<String, List<String>>, GrantedAuthority> authorityMapper =
+        record -> {
+          String role = record.get("spring.security.ldap.dn").get(0);
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java:104]: make the role a constant?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java [hunk 5]
```diff
@@ -1,45 +1,51 @@
...
 
-/**
- * Default basic configuration for taskana web example.
- */
+import pro.taskana.rest.security.SpringSecurityToJaasFilter;
+
+/** Default basic configuration for taskana web example. */
 @Configuration
 @EnableWebSecurity
 public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/common/rest/WebSecurityConfig.java:33]: What's the reason for this code duplication?

File: rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/BaseRestDocumentation.java [hunk 6]
```diff
@@ -24,6 +24,9 @@
 @Import(ResultHandlerConfiguration.class)
 public abstract class BaseRestDocumentation {
 
+  static String ADMIN_CREDENTIALS = "Basic YWRtaW46YWRtaW4=";
+  static String TEAMLEAD_1_CREDENTIALS = "Basic dGVhbWxlYWQtMTp0ZWFtbGVhZC0x";
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/doc/api/BaseRestDocumentation.java:28]: protected

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskanaRestSpringBootApplication.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package pro.taskana.rest;
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskanaRestSpringBootApplication.java:1]: wrong package

File: rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskanaRestSpringBootApplication.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class TaskanaRestSpringBootApplication {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskanaRestSpringBootApplication.java:7]: Why do wee need this Application?

**Pull request merged**